File: flutter_app/lib/models/usuario.dart
class Usuario {
  Usuario({required this.id, required this.nombre, required this.username, required this.rol});

  final int id;
  final String nombre;
  final String username;
  final String rol;

  bool get esAdmin => rol == 'admin';

  factory Usuario.fromJson(Map<String, dynamic> json) => Usuario(
        id: json['id'] as int,
        nombre: json['nombre'] as String,
        username: json['username'] as String,
        rol: json['rol'] as String,
      );
}

File: FacturaInvent2/Models/Usuario.swift
//
//  Usuario.swift
//  FacturaInvent2
//

import Foundation

struct Usuario: Codable, Identifiable, Hashable {
    let id: Int
    var nombre: String
    var username: String
    var rol: String

    var esAdmin: Bool { rol == "admin" }
}

struct LoginResponse: Codable {
    let token: String
    let usuario: Usuario
}
